integration-tools/src/main/java/uk/ac/bbsrc/tgac/miso/integration/dp5mirage/DP5MirageScanner.java

```diff
@@ -0,0 +1,270 @@
+   * @param port DP5Mirage server port
+   * @param httpClient DP5Mirage httpClient used for testing
+   * @throws IntegrationException if the server hostname or ip could not be resolved
+   */
+  public DP5MirageScanner(String host, int port, HttpClient httpClient) throws IntegrationException {
+    this.host = host;
+    this.port = port;
+    this.httpClient = httpClient;
+  }
+
+  // Helper Methods
+  private String urlStub() {
+    return String.format("http://%s:%s"+REMOTE_STUB, host,port);
+  }
+
+  /**
+   * Returns the scan result as a JsonNode
+   * The swagger documentation details out the ScanResult in the schemas section.
+   * @return
+   */
+  private JsonNode scanRackUsingDP5(Map<String,String> parameters) throws IntegrationException {
+    HttpResponse<String> response = null;
+    JsonNode scanResults = null;
+    try {
+      // JSON response from Scanner
+      response = this.getHttpClient().send(getPostParamRequest("/scan", urlStub(), parameters),
+          BodyHandlers.ofString());
+
+      // Store results
+      ObjectMapper mapper = new ObjectMapper();
+      scanResults = mapper.readTree(response.body());
+
+    } catch (IOException | InterruptedException | URISyntaxException e) {
+      throw new IntegrationException("Error communicating with the scanner", e);
+    }
+    if (response.statusCode() != 200) {
+      // Throw error of incorrect output
+      throw new IntegrationException("Error reported by the scanner");
+    }
+    return scanResults;
+  }
+
+  /**
+   * Shutdown the DP5 headless
+   */
+  public void shutDown() throws IntegrationException {
+    HttpResponse<String> response=null;
+    try {
+      response = this.getHttpClient().send(getPutParamRequest("/system/shutdown", urlStub()),
+          BodyHandlers.ofString());
+    } catch (IOException | InterruptedException | URISyntaxException e) {
+      throw new IntegrationException("Error communicating with the scanner", e);
+    }
+    assert response != null;
+    if (response.statusCode() != 200) {
+      // Throw error of incorrect output
+      throw new IntegrationException("Error reported by the scanner");
+    }
+  }
+
+  /**
+   * Performs any setup actions required to ensure that the scanner is prepared to scan a box of the correct size. Calling this method
+   * multiple times will have the same effect. Failing to call this method before {@link #getScan()} may have undesirable consequences
+   * such as scan failure or returning of old or inaccurate data. This is a blocking call and returns when the scanner is confirmed ready
+   * to scan. Failing to provide the correct box dimensions may result in inaccurate or incomplete data
+   *
+   * @param expectedRows number of rows to expect in the rack to be scanned
+   * @param expectedColumns number of columns to expect in the rack being scanned
+   * @throws IntegrationException if the scanner cannot be accessed or any hardware-specific error occurs
+   */
```

Comment:
Don't copy interface documentation to the implementation. Same for `getScan` below